CMakeLists.txt

```diff
@@ -192,26 +192,30 @@ enable_testing()
 set( TEST_LOGFILE ${PROJECT_BINARY_DIR}/reports/installcheck.log )
 file( MAKE_DIRECTORY ${PROJECT_BINARY_DIR}/reports/ )
 
+add_custom_target( update_environment
+  COMMAND bash -c "source /home/travis/build/nest/nest-simulator/result/bin/nest_vars.sh"
+  COMMENT "Preparing environment (path: ${PROJECT_BINARY_DIR}/extras/nest_vars.sh)..."
+)
+
 if ( HAVE_PYTHON )
   add_custom_target( installcheck
     COMMAND ${CMAKE_COMMAND} -E env
-      NEST_PATH=${CMAKE_INSTALL_FULL_BINDIR}
       PYTHON="${PYTHON}"
-      NEST_PYTHONPATH="${CMAKE_INSTALL_PREFIX}/${PYEXECDIR}"
       ${CMAKE_INSTALL_FULL_DATADIR}/extras/do_tests.sh --test-pynest
     WORKING_DIRECTORY "${PROJECT_BINARY_DIR}"
     COMMENT "Execute NEST's testsuite...."
   )
 else ()
   add_custom_target( installcheck
     COMMAND ${CMAKE_COMMAND} -E env
-      NEST_PATH=${CMAKE_INSTALL_FULL_BINDIR}
       ${CMAKE_INSTALL_FULL_DATADIR}/extras/do_tests.sh
     WORKING_DIRECTORY "${PROJECT_BINARY_DIR}"
     COMMENT "Execute NEST's testsuite...."
   )
 endif ()
 
+add_dependencies( installcheck update_environment )
+
```

Comment:
Beautiful!

Reply:
This makes me wonder if `installcheck` should not also depend on `install`? Currently, if you invoke `make installcheck` before `make install`,  you get a rather incomprehensible error message. And `update_environment` depends on `install`. Can we express this also explicitly here?